### PR TITLE
SmartThings - Fix update CT color.

### PR DESCRIPTION
**Pull request by m-chau**
- add error handling for translator error.
- fix the error on changing color temperature.
- updated unittest value for colour temperature it should be value between 2000 and 6500 (as indicated at [the app code](https://microsoft.visualstudio.com/DefaultCollection/WindowsAtHome/_git/WindowsAtHome?path=%2FStoreApps%2FDeviceManager%2FControls%2FDeviceDetailsTemplates.xaml&version=GBmaster&_a=contents&line=132&lineStyle=plain&lineEnd=133&lineStartColumn=37&lineEndColumn=51)
==== Update on 3/2/2017 ====
- Support dimmerSwitch as light
- Update the groovy code and mock device data.
- Fix a bug in color conversion (HSV to RGB).

### CODE REVIEW

**Comment by danielrischar** [comment]
>                 result['colorTemperature'] = resourceSchema.ct;

OCF colourChroma.ct is supposed to be in "Mired" whereas Smartthings want's Kelvin. Mired = 1000000 / Kelvin.
so resourceSchema.ct needs to be multiplied by 1million

---
Refers to: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js:297 in 4699ff8. [](commit_id = 4699ff8d382f05c35e0e8f3c581182d220b99cb8, deletion_comment = False)

File: org.opent2t.sample.lamp.superpopular/test/lampTests.js [hunk 1]
```diff
@@ -139,10 +139,10 @@ function runLampTests(settings) {
     test.serial('SetColourChroma', t => {
         return helpers.runTest(settings, t, () => {
             return OpenT2T.invokeMethodAsync(translator, SchemaName, 'getDevicesColourChroma', [deviceId]).then((initialTemperature) => {
-                return OpenT2T.invokeMethodAsync(translator, SchemaName, 'postDevicesColourChroma', [deviceId, { 'ct': 30 }]).then(() => {
+                return OpenT2T.invokeMethodAsync(translator, SchemaName, 'postDevicesColourChroma', [deviceId, { 'ct': 2500 }]).then(() => {
```
Comment (danielrischar) [inline org.opent2t.sample.lamp.superpopular/test/lampTests.js]: >return OpenT2T.invokeMethodAsync(translator, SchemaName, 'postDevicesColourChroma', [deviceId, { 'ct': 2500 }]).then(() => { [](start = 16, length = 124)

Does this break other lamp tests that use provide colourChroma resources?

Additionally, as I mentioned in my other comment.  Unites here should be Mired, not Kelvin. [http://www.oneiota.org/revisions/1757](http://www.oneiota.org/revisions/1757)

**Comment by m-chau** [comment]
>                 result['colorTemperature'] = resourceSchema.ct;

My bad, I will add the Ct conversoin.

---
In reply to: [283751876](https://github.com/openT2T/translators/pull/189#issuecomment-283751876) [](ancestors = 283751876)

---
Refers to: org.opent2t.sample.lamp.superpopular/com.smartthings.lightbulb/js/thingTranslator.js:297 in 4699ff8. [](commit_id = 4699ff8d382f05c35e0e8f3c581182d220b99cb8, deletion_comment = False)

File: org.opent2t.sample.lamp.superpopular/test/lampTests.js (hunk 1 above)
Comment (m-chau) [inline org.opent2t.sample.lamp.superpopular/test/lampTests.js]: revert changes.

---
In reply to: [104008098](https://github.com/openT2T/translators/pull/189#discussion_r104008098) [](ancestors = 104008098)

**Comment by danielrischar** [comment]
Ehhhh.... I tried to merge my small fix to Mary's PR, but there's a bit of weirdness going on with my merge of her repo into mine. https://github.com/openT2T/translators/pull/189/commits/dea0f78b88f585f6db3be0a32d86003f74fda2bf

**Comment by danielrischar** [comment]
Sorry @m-chau for mangling your branch.

**Pull request merged**